packages/contracts/contracts/engine/V3/EngineFactoryV0.sol

```diff
@@ -0,0 +1,349 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+// @dev fixed to specific solidity version for clarity and for more clear
+// source code verification purposes.
+pragma solidity 0.8.22;
+
+import {AdminACLV0} from "../../AdminACLV0.sol";
+import {IGenArt721CoreContractV3_Engine, EngineConfiguration} from "../../interfaces/v0.8.x/IGenArt721CoreContractV3_Engine.sol";
+import {ICoreRegistryV1} from "../../interfaces/v0.8.x/ICoreRegistryV1.sol";
+import {IEngineFactoryV0} from "../../interfaces/v0.8.x/IEngineFactoryV0.sol";
+import {IAdminACLV0} from "../../interfaces/v0.8.x/IAdminACLV0.sol";
+
+import "@openzeppelin-5.0/contracts/access/Ownable.sol";
+import {Clones} from "@openzeppelin-5.0/contracts/proxy/Clones.sol";
+import {IERC20} from "@openzeppelin-5.0/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin-5.0/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Create2} from "@openzeppelin-5.0/contracts/utils/Create2.sol";
+
+/**
+ * @title EngineFactoryV0
+ * @author Art Blocks Inc.
+ * @notice Factory contract for creating new Engine and Engine Flex Core contracts.
+ * @dev This contract is deployed once, and then used to create new Engine and Engine
+ * Flex Core contracts. The contract may be abandoned once it is no longer needed.
+ * Once abandoned, the contract can no longer be used to create new Engine and Engine
+ * Flex Core contracts.
+ * The contract is initialized with a required contract type.
+ * The contract is initialized with an Engine and Engine Flex implementation contract, which is cloned
+ * when creating new Engine and Engine Flex Core contracts.
+ */
+contract EngineFactoryV0 is Ownable, IEngineFactoryV0 {
+    // public type
+    bytes32 public constant type_ = "EngineFactoryV0";
+
+    /**
+     * The implementation contract that is cloned when creating new Engine
+     * contracts.
+     */
+    address public immutable engineImplementation;
+    /**
+     * The implementation contract that is cloned when creating new Engine
+     * Flex contracts.
+     */
+    address public immutable engineFlexImplementation;
+
+    // Address of core registry contract.
+    address public immutable coreRegistry;
+
+    /// version and type of Engine implementation contract
+    string public coreType;
+    string public coreVersion;
```

Comment:
considering there are also `flexCoreType`/version, wdyt about naming these something like `engineCoreType`/version? idk if that is better or not, but just wanted to double click

Reply:
good call! updated this to reflect `engineCoreType/version`